### PR TITLE
Updated tests to support latest avvio

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #334.

### CODE REVIEW

**Comment by delvedor** [comment]
Should we update also the documentation?
I'll add also the following test:
```js
test('Plugin with multiple prefix', t => {
  t.plan(2)
  const fastify = Fastify()

  fastify.register(parent, { prefix: '/parent' })

  function parent (instance, opts, next) {
    instance.register(child, { prefix: '/child1' })
    instance.register(child, { prefix: '/child2' })
    next()
  }

  function child (instance, opts, next) {
    instance.get('/', (req, reply) => {
      reply.send({ prefix: opts.prefix })
    })
    next()
  }

  fastify.inject({
    method: 'GET',
    url: '/parent/child1'
  }, res => {
    t.same(JSON.parse(res.payload), { prefix: '/child1' })
  })

  fastify.inject({
    method: 'GET',
    url: '/parent/child2'
  }, res => {
    t.same(JSON.parse(res.payload), { prefix: '/child2' })
  })
})
```

File: test/plugin.test.js [hunk 1]
```diff
@@ -349,3 +349,49 @@ test('add hooks after route declaration', t => {
...
+
+    fastify.register((fastify, opts, next) => {
+      fastify.get('/', function (req, reply) {
+        reply.send('I am child 2')
+      })
+      next()
+    }, { prefix: '/child2' })
+
+    next()
+  })
```
Comment (mcollina) [inline test/plugin.test.js]: @delvedor this test should be the one that you referring to.

Comment (delvedor) [reply test/plugin.test.js]: Yes, it is!
I would like to add another prefix (`/parent`) to be super sure about the parent's prefix as well.

**Comment by mcollina** [comment]
@delvedor done.

**Comment by mcollina** [comment]
I will push a fix for the docs later on.

File: test/internals/reply.test.js [hunk 2]
```diff
@@ -92,161 +41,146 @@ test('reply.serializer should set a custom serializer', t => {
...
-      reply.redirect('/')
-    })
-    t.pass()
-  } catch (e) {
-    t.fail()
-  }
-})
+  fastify.get('/', function (req, reply) {
+    reply.code(200)
+    reply.header('Content-Type', 'text/plain')
```
Comment (allevo) [inline test/internals/reply.test.js]: Could you use `type` as below?

Comment (mcollina) [reply test/internals/reply.test.js]: that's the point of the test, exercising `.header()`.

File: test/internals/reply.test.js [hunk 3]
```diff
@@ -92,161 +41,146 @@ test('reply.serializer should set a custom serializer', t => {
...
-    uri: 'http://localhost:' + fastify.server.address().port + '/undefined'
-  }, (err, response, body) => {
+  fastify.listen(0, err => {
     t.error(err)
-  })
+    fastify.server.unref()
+
+    test('custom serializer should be used', t => {
+      t.plan(3)
+      request({
```
[the remark this answers is not in the document]
Comment (mcollina) [reply test/internals/reply.test.js]: These tests were just refactored. I would prefer to leave them as they are.

**Pull request merged**